sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -452,27 +446,26 @@ def query_items(
             response_hook(self.client_connection.last_response_headers, items)
         return items
 
-    def __is_prefix_partitionkey(self, partition_key: Union[str, list]):
+    def __is_prefix_partitionkey(self, partition_key: Union[str, int, bool, float, Sequence[Any]]):
         properties = self._get_properties()
         pk_properties = properties["partitionKey"]
         partition_key_definition = PartitionKey(path=pk_properties["paths"], kind=pk_properties["kind"])
         if partition_key_definition.kind != "MultiHash":
             return False
-        if type(partition_key) == list and len(partition_key_definition['paths']) == len(partition_key):
+        if isinstance(partition_key, Sequence) and len(partition_key_definition['paths']) == len(partition_key):
             return False
         return True
 
     @distributed_trace
-    def replace_item(
+    def replace_item(  # pylint:disable=docstring-missing-param
         self,
-        item,  # type: Union[str, Dict[str, Any]]
-        body,  # type: Dict[str, Any]
-        populate_query_metrics=None,  # type: Optional[bool] # pylint:disable=docstring-missing-param
-        pre_trigger_include=None,  # type: Optional[str]
-        post_trigger_include=None,  # type: Optional[str]
-        **kwargs  # type: Any
-    ):
-        # type: (...) -> Dict[str, Any]
+        item: Union[str, Mapping[str, Any]],
+        body: Mapping[str, Any],
```

Comment:
I need to do some tests here to confirm that `Mapping` is actually supported all the way down the stack, or whether it should in fact be `Dict`.

Reply:
This should now be fixed.